packages/core/src/crawlers/crawler_commons.ts

```diff
@@ -81,3 +81,17 @@ export interface CrawlingContext<Crawler = unknown, UserData extends Dictionary
      */
     sendRequest<Response = string>(overrideOptions?: Partial<OptionsInit>): Promise<GotResponse<Response>>;
 }
+
+export interface RestrictedCrawlingContext<UserData extends Dictionary = Dictionary> {
+    request: Request<UserData>;
+    pushData: (record: Dictionary, datasetName?: string) => Promise<void>;
+    enqueueLinks: (options?: EnqueueLinksOptions) => Promise<void>;
+    addRequests: (
+        requestsLike: Source[],
+        options?: RequestQueueOperationOptions,
+    ) => Promise<void>;
+    getStoreValue: <Value = unknown>(key: string, defaultValue?: Value) => Promise<Value>
```

Comment:
Well, downloading files, making screenshots and the like may be valid usecases for this that I feel are not really suitable for `useState`.

Regarding the naming, I'm not thrilled with this either. How about `get/setKeyedValue`?

Reply:
i'd prefer `getValue` and `setValue`, feels in line with our current `pushData` method, which is alias for `Dataset.pushData`, and those would alias `KeyValueStore.getValue` and `KeyValueStore.setValue` respectively

naming is hard, thoughts @barjin?

Reply:
Those are also a possibility for me. I'd like the name to convey that it uses the key-value store, but I can't judge how important that really is.

Reply:
`(get|set)Value` would be a no-brainer for me here, exactly because of `KVS` methods... but trying to see this through newcomer's eyes, it does get slightly confusing (`setValue` / `pushData`).

imo consistency is the key here though, anything I tried coming up with (`KVSgetValue` etc.) seems worse to me... and passing the entire default KVS object is even more inconsistent with what we put in the context (helper methods).

Reply:
we could make a restricted key-value store client and return that :shrug: